klient/decoder/decoder.go

```diff
@@ -55,6 +56,18 @@ type MutateFunc func(obj k8s.Object) error
 // HandlerFunc is a function executed after an object has been decoded and patched. If an error is returned, further decoding is halted.
 type HandlerFunc func(ctx context.Context, obj k8s.Object) error
 
+// PreHandlerFuncHook is a handler helper provided to hook into the Workflow of HandlerFunc to make sure you can control what resources
+// are installed programatically in case if required. This can help in performing controlled CRUD operaton of the k8s resource using the
+// decoder package
+type PreHandlerFuncHook func(ctx context.Context, obj k8s.Object) bool
```

Comment:
This seems to be adding yet another level of control when accessing the flow of decoding/encoding execution.  The `decoder` functions already provide ways of controlling what resources are selected via the glob patterns.  Can you give an example when you think this would be needed ?

Reply:
The idea behind this was to be able to provide a way for end users to programatically skip certain resource discovered.  
But then again the same can be done via the Pattern regex as well. Should I keep this or do you think it is better to cleanup this ?

Say I build the code once but I want to control what resource gets processed via the test, I can use this conditional check as a way to control that. But I might be over thinking this.

Reply:
@harshanarayana LoL.. I think you are overthinking this one. 
Remember, a test user can specify the name or name patterns for the resource file.
I just think that this is a bit overkill, specially  when I can specify the file name or a name pattern to select the resource file from a directory.

Reply:
:) I have cleaned it up.